source/nodejs/tests/ui-tests/src/card-element-utils.ts

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { WebElement, By } from "selenium-webdriver";
+import { TestUtils } from "./test-utils";
+import * as Assert from "assert";
+
+export class Element {
+    underlyingElement?: WebElement = undefined;
+
+    protected constructor() {}
+
+    elementWasFound(): boolean
+    {
+        return (this.underlyingElement !== undefined);
+    }
+}
+
+export abstract class Input extends Element {
+    protected constructor() {
+        super();
+    }
+
+    abstract setData(data: string): Promise<void>;
+}
+
+export class InputText extends Input {
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputText>
+    {
+        let input = new InputText();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-textInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void>
+    {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+
+    async isFocused(): Promise<boolean>
+    {
+        if (this.elementWasFound())
+        {
+            const inputId: string = await this.underlyingElement!.getAttribute("id");
+            const activeElementId: string = await TestUtils.getInstance().getDriver().switchTo().activeElement().getAttribute("id");
+
+            return (inputId === activeElementId);
+        }
+
+        return false;
+    }
+}
+
+export class InputDate extends Input {
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputDate>
+    {
+        let input = new InputDate();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-dateInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void> 
+    {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+}
+
+export class InputTime extends Input {
```

Comment:
I don't know why you are create some many classes. can you just pass a id to Input or Element constructor?

In Input, it looks like this:
```
constructor(id: string, type: string) {
this.id = id;
this.type = type;}
get underlyingElement() { return TestUtils.getInstanace().getInput(this.id, this.type}
```

and you may have help function like:
```
 createInputTime(id: string) { return new Input(id, "ac-timeInput")}
```